service/history/workflow/update/update.go

```diff
@@ -271,6 +271,11 @@ func (u *Update) abort(reason AbortReason) {
 
 	const preCompletedStates = stateSet(stateAccepted | stateProvisionallyCompleted)
 	if u.state.Matches(preCompletedStates) {
+		// Accepted Update can't be applied to the new run, and must to be aborted
+		// same way as if Workflow is completed.
+		if reason == AbortReasonWorkflowContinuing {
+			reason = AbortReasonWorkflowCompleted
+		}
```

Comment:
Could we change this to pick a different `error` instead? Changing the `reason` feels like the abstraction doesn't work since we're patching the reason based on the Update's state. I think the abort reason is like an "intent", and the `error` is the actual decision.

ie
```
var abortError err
if reason == AbortReasonWorkflowContinuing {
    ...
} else {
    ...
}
```

Reply:
I like you notice this. I don't like it either. I changed it to error substitution. Still not great, but, I agree, slightly better.